debian/control

```diff
@@ -934,6 +937,41 @@ Description: CrackLib Password Validation Plugin for MariaDB
  .
  Install and configure this to enforce stronger passwords for MariaDB users.
 
+Package: mariadb-plugin-provider-bzip2
```

Comment:
Why are these 4 different packages? Seems a bit over-engineered. Why not include those 4 .so libraries in the server (with current code they are all build anyway) and then just let users choose what compression method they want to use via a config option?

Reply:
This is the whole purpose of this task. The user decides what compression methods they wants and will only need to install those libraries. There is no need to install libraries one doesn't use.

The whole issue of "do benchmarks" and "not enable all compression methods" was about a choice what run-time dependencies to enable to satisfy as many users as possible, but without adding too many dependencies.

And this was, generally, impossible to solve. Because nobody uses or needs *all* compression libraries, but for every compression library there was somebody who needed it. That is, everybody — every single user — was forced to install libraries they didn't need only because someone in the world might've needed them.

Now we've solved it, a user only installs libraries they want to use.

Reply:
1) Maybe plugin name should be mariadb-plugin-compression-bzip2 ? Would better indicate what it is about. For an average DBA 'provider' is very inspecific.

2) Do we have a CI that has built these .deb packages or do I need to rebuild this myself? I am planning to check what dependencies the packages actually end up with. If the goal is to avoid unnecessary dependencies, we need to verify the list of the dependencies of the server before and after this change.

Reply:
1. Debian pattern is `mariadb-plugin-<plugin name>`, for example `mariadb-plugin-rocksdb`. The plugin is called "provider_bzip2" because in the future we plan to use it for more then just compression. I was hoping that "bzip2" part is specific enough, and error messages saying "you're trying to use bzip2 compression, but its provider plugin is not installed" will also help.

2. Absolutely. Packages are here: http://hasky.askmonty.org/archive/bb-10.7-release/build-42268/

Reply:
1. All of the plugins provide something. I simply think that the term 'provider' is redundant here. Shorter would be sweeter. But as they say, naming is one of the hard things in computer science..
2. Ok, I have verified that the dependencies are correctly separated.